src/overlays/actors/ovl_Item_Etcetera/z_item_etcetera.c

```diff
@@ -26,29 +35,186 @@ const ActorInit Item_Etcetera_InitVars = {
+        }
+        Actor_Kill(&this->actor);
+    } else {
+        if (0) {} // Necessary to match
+        func_8002F434(&this->actor, globalCtx, this->getItemId, 30.0f, 50.0f);
+        if ((globalCtx->gameplayFrames & 0xD) == 0) {
+            func_800293E4(globalCtx, &this->actor.posRot.pos, 0.0f, 0.0f, 10.0f, 0.13f);
+        }
+    }
+}
+
+void func_80B8598C(ItemEtcetera* this, GlobalContext* globalCtx) {
+
+    static Vec3f D_80B85D74 = { 0.0f, 0.2f, 0.0f };
+    static Vec3f D_80B85D80 = { 0.0f, 0.05f, 0.0f };
+
+    static Color_RGB8 D_80B85D8C = { 255, 255, 255 };
+    static Color_RGB8 D_80B85D90 = { 255, 50, 50 };
+
+    Vec3f vec;
+
+    D_80B85D74.x = Math_Rand_CenteredFloat(3.0f);
+    D_80B85D74.z = Math_Rand_CenteredFloat(3.0f);
+    D_80B85D74.y = -0.05f;
+    D_80B85D80.y = -0.025f;
+    vec.x = Math_Rand_CenteredFloat(12.0f) + this->actor.posRot.pos.x;
+    vec.y = (Math_Rand_ZeroOne() * 6.0f) + this->actor.posRot.pos.y;
+    vec.z = Math_Rand_CenteredFloat(12.0f) + this->actor.posRot.pos.z;
+    func_80028BB0(globalCtx, &vec, &D_80B85D74, &D_80B85D80, &D_80B85D8C, &D_80B85D90, 0x1388, 0x10);
+}
+
+void ItemEtcetera_MoveFireArrowDown(ItemEtcetera* this, GlobalContext* globalCtx) {
+    func_8002E4B4(globalCtx, &this->actor, 10.0f, 10.0f, 0.0f, 5);
+    Actor_MoveForward(&this->actor);
+    if (!(this->actor.bgCheckFlags & 1)) {
+        func_80B8598C(this, globalCtx);
+    }
+    this->actor.shape.rot.y += 0x400;
+    func_80B85824(this, globalCtx);
+}
+
+void func_80B85B28(ItemEtcetera* this, GlobalContext* globalCtx) {
+    if (Flags_GetTreasure(globalCtx, (this->actor.params >> 8) & 0x1F)) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void ItemEtcetera_UpdateFireArrow(ItemEtcetera* this, GlobalContext* globalCtx) {
+    if ((globalCtx->csCtx.state != 0) && (globalCtx->csCtx.npcActions[0] != NULL)) {
+        LOG_NUM("(game_play->demo_play.npcdemopnt[0]->dousa)", globalCtx->csCtx.npcActions[0]->action,
+                "../z_item_etcetera.c", 441);
+        if (globalCtx->csCtx.npcActions[0]->action == 2) {
+            this->actor.draw = ItemEtcetera_DrawFireArrow;
+            this->actor.gravity = -0.1f;
+            this->actor.minVelocityY = -4.0f;
+            this->actionFunc_15C = ItemEtcetera_MoveFireArrowDown;
+        }
+    } else {
+        this->actor.gravity = -0.1f;
+        this->actor.minVelocityY = -4.0f;
+        this->actionFunc_15C = ItemEtcetera_MoveFireArrowDown;
+    }
+}
+
+void ItemEtcetera_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ItemEtcetera* this = THIS;
+    this->actionFunc_15C(this, globalCtx);
+}
+
+void func_80B85C64(Actor* thisx, GlobalContext* globalCtx) {
+    ItemEtcetera* this = THIS;
+    if (globalCtx->actorCtx.unk_03 != 0) {
+        func_8002EBCC(&this->actor, globalCtx, 0);
+        func_8002ED80(&this->actor, globalCtx, 0);
+        func_800694A0(globalCtx, this->drawId);
+    }
+}
+
+void ItemEtcetera_DrawFireArrow(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
This function is used for all non-cmg items, not just fire arrows, so I suggest simply calling it "Draw", to go along with the other draw function.
```suggestion
void ItemEtcetera_Draw(Actor* thisx, GlobalContext* globalCtx) {
```